### PR TITLE
Add form validation

### PR DESCRIPTION
**Pull request by tlorent**

### CODE REVIEW

File: components/new-timesheet/NewTimesheet.jsx [hunk 1]
```diff
@@ -71,14 +99,17 @@ class NewTimeEntry extends React.Component {
...
+          new-time-entry
+          new-time-entry--${isVisible ? 'visible' : 'invisible'}
+        `}
         >
           <h2 className="new-time-entry__title">
             New time entry
           </h2>
-          <form onSubmit={handleSubmit}>
+          <form
+            onSubmit={handleSubmit}
```
Comment (RobbinHabermehl) [inline components/new-timesheet/NewTimesheet.jsx]: It's a bit overkill to put the prop on a single line if the element just has one prop.

File: components/new-timesheet/NewTimesheet.jsx [hunk 2]
```diff
@@ -93,11 +124,16 @@ class NewTimeEntry extends React.Component {
                 >
                   Employer
                   <input
-                    className="new-time-entry__input-field new-time-entry__input-field--large"
+                    className={`new-time-entry__input-field new-time-entry__input-field--${formInputValidity.employer ? '' : 'invalid'} new-time-entry__input-field--large`}
```
Comment (RobbinHabermehl) [inline components/new-timesheet/NewTimesheet.jsx]: With more than one class name the readability will be increased by putting all class names on a dedicated line, like so:

```
<input
  className={`
    new-time-entry__input-field
    new-time-entry__input-field--${....}
  `}
```

File: components/new-timesheet/NewTimesheet.jsx [hunk 3]
```diff
@@ -155,20 +205,28 @@ class NewTimeEntry extends React.Component {
...
                   To
                   <input
-                    className="new-time-entry__input-field new-time-entry__input-field--small"
+                    className={`new-time-entry__input-field new-time-entry__input-field--${formInputValidity.stopTime ? '' : 'invalid'} new-time-entry__input-field--small`}
                     type="text"
+                    required
                     name="stopTime"
                     id="stopTime"
+                    placeholder="00:00"
+                    pattern="^(?:[01]?[0-9]|2[0-3])[./:][0-5][0-9]$"
```
Comment (RobbinHabermehl) [inline components/new-timesheet/NewTimesheet.jsx]: Nice!

File: components/new-timesheet/NewTimesheet.jsx [hunk 4]
```diff
@@ -7,18 +7,26 @@ import './new-timesheet.scss';
...
+      project: true,
+      startTime: true,
+      stopTime: true
+    },
+    isVisible: false
   }
 
   state = { ...NewTimeEntry.defaultState }
 
+  checkFormValidity = () => Object.values(this.state.formInputValidity).every((value) => value)
```
Comment (RobbinHabermehl) [inline components/new-timesheet/NewTimesheet.jsx]: Nicely implemented like this!

File: components/new-timesheet/NewTimesheet.jsx [hunk 5]
```diff
@@ -155,20 +205,28 @@ class NewTimeEntry extends React.Component {
...
                   <input
-                    className="new-time-entry__input-field new-time-entry__input-field--small"
+                    className={`new-time-entry__input-field new-time-entry__input-field--${formInputValidity.stopTime ? '' : 'invalid'} new-time-entry__input-field--small`}
                     type="text"
+                    required
                     name="stopTime"
                     id="stopTime"
+                    placeholder="00:00"
+                    pattern="^(?:[01]?[0-9]|2[0-3])[./:][0-5][0-9]$"
+                    title="00:00"
```
Comment (larstroost) [inline components/new-timesheet/NewTimesheet.jsx]: title could be HH:MM as well, to specify to order, although that should be clear when using time

File: components/new-timesheet/NewTimesheet.jsx [hunk 6]
```diff
@@ -93,11 +124,16 @@ class NewTimeEntry extends React.Component {
                 >
                   Employer
                   <input
-                    className="new-time-entry__input-field new-time-entry__input-field--large"
+                    className={`new-time-entry__input-field new-time-entry__input-field--${formInputValidity.employer ? '' : 'invalid'} new-time-entry__input-field--large`}
                     type="text"
+                    required
+                    minLength="2"
+                    maxLength="30"
```
Comment (larstroost) [inline components/new-timesheet/NewTimesheet.jsx]: What about the company: Meubelstoffeerderij De Stofwisseling. I don't think it is necessary to add a max length

**Pull request merged**